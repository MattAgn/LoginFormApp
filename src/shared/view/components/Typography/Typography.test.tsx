import { screen } from '@testing-library/react-native';

import { renderWithProviders } from '#testing/render';

import { Typography } from './Typography';

describe('Typography', () => {
  test('Title xxxlarge matches snapshot', () => {
    renderWithProviders(
      <Typography.TitleXXXLarge>hello</Typography.TitleXXXLarge>,
    );

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title xxlarge matches snapshot', () => {
    renderWithProviders(
      <Typography.TitleXXLarge>hello</Typography.TitleXXLarge>,
    );

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title xlarge matches snapshot', () => {
    renderWithProviders(<Typography.TitleXLarge>hello</Typography.TitleXLarge>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title large matches snapshot', () => {
    renderWithProviders(<Typography.TitleLarge>hello</Typography.TitleLarge>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title medium matches snapshot', () => {
    renderWithProviders(<Typography.TitleMedium>hello</Typography.TitleMedium>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title small matches snapshot', () => {
    renderWithProviders(<Typography.TitleSmall>hello</Typography.TitleSmall>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('Title xsmall matches snapshot', () => {
    renderWithProviders(<Typography.TitleXSmall>hello</Typography.TitleXSmall>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P1 regular matches snapshot', () => {
    renderWithProviders(<Typography.P1Regular>hello</Typography.P1Regular>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P1 bold matches snapshot', () => {
    renderWithProviders(<Typography.P1Bold>hello</Typography.P1Bold>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P1 italic matches snapshot', () => {
    renderWithProviders(<Typography.P1Italic>hello</Typography.P1Italic>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P2 regular matches snapshot', () => {
    renderWithProviders(<Typography.P2Regular>hello</Typography.P2Regular>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P2 bold matches snapshot', () => {
    renderWithProviders(<Typography.P2Bold>hello</Typography.P2Bold>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P2 italic matches snapshot', () => {
    renderWithProviders(<Typography.P2Italic>hello</Typography.P2Italic>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P3 regular matches snapshot', () => {
    renderWithProviders(<Typography.P3Regular>hello</Typography.P3Regular>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P3 bold matches snapshot', () => {
    renderWithProviders(<Typography.P3Bold>hello</Typography.P3Bold>);

    expect(screen).toMatchComponentSnapshot();
  });

  test('P3 italic matches snapshot', () => {
    renderWithProviders(<Typography.P3Italic>hello</Typography.P3Italic>);

    expect(screen).toMatchComponentSnapshot();
  });
});
