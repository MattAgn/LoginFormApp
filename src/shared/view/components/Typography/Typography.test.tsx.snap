// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography P1 bold matches snapshot 1`] = `"hello"`;

exports[`Typography P1 bold matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 16,
        "lineHeight": 24,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P1 italic matches snapshot 1`] = `"hello"`;

exports[`Typography P1 italic matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Italic",
        "fontSize": 16,
        "lineHeight": 24,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P1 regular matches snapshot 1`] = `"hello"`;

exports[`Typography P1 regular matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
        "lineHeight": 24,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P2 bold matches snapshot 1`] = `"hello"`;

exports[`Typography P2 bold matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 14,
        "lineHeight": 21,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P2 italic matches snapshot 1`] = `"hello"`;

exports[`Typography P2 italic matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Italic",
        "fontSize": 14,
        "lineHeight": 21,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P2 regular matches snapshot 1`] = `"hello"`;

exports[`Typography P2 regular matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 14,
        "lineHeight": 21,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P3 bold matches snapshot 1`] = `"hello"`;

exports[`Typography P3 bold matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 12,
        "lineHeight": 18,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P3 italic matches snapshot 1`] = `"hello"`;

exports[`Typography P3 italic matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Italic",
        "fontSize": 12,
        "lineHeight": 18,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography P3 regular matches snapshot 1`] = `"hello"`;

exports[`Typography P3 regular matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    numberOfLines={0}
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 12,
        "lineHeight": 18,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title large matches snapshot 1`] = `"hello"`;

exports[`Typography Title large matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 30,
        "lineHeight": 39,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title medium matches snapshot 1`] = `"hello"`;

exports[`Typography Title medium matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 26,
        "lineHeight": 34,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title small matches snapshot 1`] = `"hello"`;

exports[`Typography Title small matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 22,
        "lineHeight": 29,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title xlarge matches snapshot 1`] = `"hello"`;

exports[`Typography Title xlarge matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 32,
        "lineHeight": 42,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title xsmall matches snapshot 1`] = `"hello"`;

exports[`Typography Title xsmall matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 18,
        "lineHeight": 23,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title xxlarge matches snapshot 1`] = `"hello"`;

exports[`Typography Title xxlarge matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 36,
        "lineHeight": 47,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;

exports[`Typography Title xxxlarge matches snapshot 1`] = `"hello"`;

exports[`Typography Title xxxlarge matches snapshot 2`] = `
<RNCSafeAreaProvider
  style={
    {
      "flex": 1,
    }
  }
>
  <Text
    accessibilityRole="header"
    style={
      {
        "color": "#000000",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 46,
        "lineHeight": 60,
        "textDecorationLine": "none",
      }
    }
  >
    hello
  </Text>
</RNCSafeAreaProvider>
`;
